cscs-checks/cuda/cuda_checks.py

```diff
@@ -1,61 +1,69 @@
 import os
 
+import reframe as rfm
 import reframe.utility.sanity as sn
-from reframe.core.pipeline import RegressionTest
 
 
-class CudaCheck(RegressionTest):
-    def __init__(self, name, **kwargs):
-        super().__init__('cuda_%s_check' % name,
-                         os.path.dirname(__file__), **kwargs)
+class CudaCheck(rfm.RegressionTest):
+    def __init__(self, name):
+        super().__init__()
+        self.name = 'cuda_%s_check' % name
         self.valid_systems = ['daint:gpu', 'dom:gpu', 'kesch:cn']
         self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu']
         self.sourcesdir = os.path.join(self.current_system.resourcesdir,
                                        'CUDA', 'essentials')
         self.modules = ['craype-accel-nvidia60']
-        self.maintainers = ['AJ', 'VK']
         self.num_gpus_per_node = 1
-        self.tags = {'production'}
-
-    def compile(self):
-        # Set nvcc flags
-        nvidia_sm = '60'
+        self.nvidia_sm = '60'
         if self.current_system.name == 'kesch':
-            nvidia_sm = '37'
-        self.current_environ.cxxflags = ('-ccbin g++ -m64 -lcublas '
-                                         '-arch=sm_%s' % nvidia_sm)
-        super().compile()
+            self.nvidia_sm = '37'
 
+        self.maintainers = ['AJ', 'VK']
+        self.tags = {'production'}
 
+
+@rfm.simple_test
 class MatrixmulCublasCheck(CudaCheck):
```

Comment:
You can prefix all subclasses with `Cuda`, so as to get automatically the correct names.